bin/execution.ml

```diff
@@ -30,6 +30,25 @@ let char_to_stack_int char =
 (* Converts a char into an integer*)
 let char_to_int char = int_of_char char - int_of_char '0'
 
+
+let epsilon = 0.0001
+let compare_float v u =
+  let diff = abs_float (v -. u) in
+  if abs_float v > 1.0 || abs_float u > 1.0 then
+    if diff <= epsilon *. max (abs_float v) (abs_float u) then 0 else 1
+  else if diff <= epsilon then 0
+  else 1
```

Comment:
I think this is not correct.

> When comparing two floating-point numbers, these numbers are regarded
as equal if they differ at most by epsilon (if both numbers
are in the range between -1.0 and 1.0) or epsilon multiplied
by the larger absolute value of the two numbers (if the absolute value of at least one of the numbers is larger than 1.0).

and in you case above 
(abs_float (1.0 -. -1.0)) will give 2 back which will go in the first if (but it hsould go in the else)

Reply:
Depends if the range includes or excludes 1.0 and -1.0
If it includes both it can be fixed by using v >= 1.0 and u >= 1.0

The first calculation does not have to do anything with the first if, its only the diff between the two values